x-pack/test/functional/apps/apm/correlations/failed_transaction_correlations.ts

```diff
@@ -0,0 +1,155 @@
+        });
+      });
+
+      it(`navigates to the 'opbeans-go' service overview page`, async function () {
+        await find.clickByDisplayedLinkText(testData.serviceName);
+
+        await retry.try(async () => {
+          const apmMainTemplateHeaderServiceName = await testSubjects.getVisibleTextAll(
+            'apmMainTemplateHeaderServiceName'
+          );
+          expect(apmMainTemplateHeaderServiceName).to.contain(testData.serviceName);
+        });
+      });
+
+      it('navigates to the transactions tab', async function () {
+        await find.clickByDisplayedLinkText(testData.transactionsTab);
+
+        await retry.try(async () => {
+          const apmMainContainerText = await testSubjects.getVisibleTextAll('apmMainContainer');
+          const apmMainContainerTextItems = apmMainContainerText[0].split('\n');
+
+          expect(apmMainContainerTextItems).to.contain(testData.transaction);
+        });
+      });
+
+      it(`navigates to the 'GET /api/stats' transactions`, async function () {
+        await find.clickByDisplayedLinkText(testData.transaction);
+
+        await retry.try(async () => {
+          const apmMainContainerText = await testSubjects.getVisibleTextAll('apmMainContainer');
+          const apmMainContainerTextItems = apmMainContainerText[0].split('\n');
+
+          expect(apmMainContainerTextItems).to.contain(testData.transaction);
+          expect(apmMainContainerTextItems).to.contain(testData.correlationsTab);
+        });
+      });
+
+      it('shows the failed transactions correlations tab', async function () {
+        await testSubjects.click('apmFailedTransactionsCorrelationsTabButton');
+
+        await retry.try(async () => {
+          await testSubjects.existOrFail('apmFailedTransactionsCorrelationsTabContent');
+        });
+      });
+
+      it('loads the failed transactions correlations results', async function () {
+        await retry.try(async () => {
+          const apmFailedTransactionsCorrelationsTabTitle = await testSubjects.getVisibleText(
+            'apmFailedTransactionsCorrelationsTabTitle'
+          );
+          expect(apmFailedTransactionsCorrelationsTabTitle).to.be('Failed transactions');
+
+          // Assert that the data fully loaded to 100%
+          const apmFailedTransactionsCorrelationsProgressTitle = await testSubjects.getVisibleText(
+            'apmCorrelationsProgressTitle'
+          );
+          expect(apmFailedTransactionsCorrelationsProgressTitle).to.be('Progress: 100%');
+
+          // Assert that results for the given service didn't find any correlations
+          const apmCorrelationsTable = await testSubjects.getVisibleText('apmCorrelationsTable');
+          expect(apmCorrelationsTable).to.be(
+            'No significant correlations\nCorrelations will only be identified if they have significant impact.\nTry selecting another time range or remove any added filter.'
+          );
+        });
+      });
+    });
+  });
+}
```

Comment:
Thanks for adding this test! Do we already have API tests for the correlations endpoint? If not, can you please add one? Thanks!

Reply:
Added the test for the failed transaction correlations search strategy here https://github.com/elastic/kibana/pull/109187/commits/e683e57ee486a81512223cff873b35730b0346a3